new text on idle timeouts (#440)
quicwg/multipath: draft-ietf-quic-multipath.md

When only one path is available, servers MUST follow the specifications
in {{QUIC-TRANSPORT}}.

> This dcoument does not specify any specific per-path timeouts. An endpoint

Author: ```suggestion
This document does not specify any specific per-path timeouts. An endpoint
```

> can decide to close a path at any time not matter if the path is in active
> use or not by sending a PATH_ABANDON frame. Particulary it is not required

Reviewer: ```suggestion
can decide to close a path at any time, whether the path is in active
use or not, by sending a PATH_ABANDON frame. Particularly, it is not required
```

to send a PATH_ABANDON frame at any specific point in time if that would be

> the only frame to send.

Reviewer 2: Typo dcoument 

I think we could rewrite the second sentence a little bit. Maybe: 
"It is not required to send a PATH_ABANDON frame at any specific point in time. For example,
the endpoint may want to wait for until it is not the only frame to send."

> use or not, by sending a PATH_ABANDON frame. Particularly, it is not required
> to send a PATH_ABANDON frame at any specific point in time if that would be
> the only frame to send.

Author: ```suggestion
use or not, by sending a PATH_ABANDON frame. It is not required
to send a PATH_ABANDON frame at any specific point in time if that would be
the only frame to send. For example, an endpoint may want to wait until
it will anyway send another frame.
```

> For example, an endpoint may wait until it will anyway send another frame.

Reviewer 3: > It is not required to send a PATH_ABANDON frame at any specific point in time.

Do you mean "An endpoint does not need to send a PATH_ABANDON frame immediately or at any particular moment"?

Author: yes, one endpoint decides to close a path and then need to send an abandon frame, however, there is nothing in this spec that says it has to take that decision as an specific point in time, e.g. as react to a certain event. For example even if you receive a packet on a new path that you don't want to open, it doesn't necessarily mean that you have to close it immediately, because opening a path (by receiving) a packet is an independent event from closing the path, meaning as you as you have received the path it is effetely on open path and similarly as you can decide to close any path at any point of time, you simply also decide to close that path short after opening it. Sorry for using these many works but I hope that makes my mental model more clear....?

If a path is not actively used for a while, it might not be usable anymore,
e.g. due to middlebox timeouts. To avoid such path breakage, endpoints
can send ack-eliciting packets such as packets containing PING frames
({{Section 19.2 of QUIC-TRANSPORT}}) on that path to keep it alive.
As discussed in
{{Section 10.1.2 of QUIC-TRANSPORT}}, the keep-alive interval depends
on the timeout in the middlebox.

> If a path was not actively used for a while, an endpoint can

Author: We switched here from "might consider" to "can". However, with "can" the sentence seems incorrect to me as we are not giving an advise anymore. So yes, the endpoint can do it but why are we talking about it...? I'm not a native speaker but I would think "might" would make slightly more sense...? Or do we need to re-phrase this entirely?

Reviewer 2: I think this is OK. We might say "endpoints could send", to emphasize that this is some optional behavior. We certainly don't want to make a specific recommendation like 'send every 15 seconds', because the behavior of middleboxes is unpredictable.

> If a path was not actively used for a while, an endpoint might consider
> to probe it before switching to active use if there are still other paths

Reviewer: ```suggestion
If a path was not actively used for a while, an endpoint can
probe it before switching to active use if there are still other paths
```
